actioncable/lib/action_cable/connection/stream_event_loop.rb

```diff
@@ -5,6 +5,8 @@
 module ActionCable
   module Connection
     class StreamEventLoop
+      include ActiveSupport::Inspect({ "attached" => -> { @map.size }, "loop" => :@thread, "post_queue" => -> { @executor&.queue_length }, "post_threads" => -> { @executor&.length } })
```

Comment:
What about adding some syntactic sugar to avoid having to include the module constructor.

For example:

```ruby
class SomeClass
  inspect_dump do |dump|
    dump.label "Some label!"
    dump.attached @map.size
    dump.loop @thread
    dump.post_queue @executor&.queue_length
    dump.post_threads @executor&.length
  end
end
```

Internally the class method could include the module constructor. I think this would be slightly nicer to read.

Also, I think it would be desirable to feel that you are writing a regular instance method when configuring the output. So for example, use `:@thread` instead of `@thread`. This should be possible by evaluating the block with `instance_exec` on the instance and passing both the properties (`dump`) object and the block to configure.

I would try to do that independently of whether we add more syntax: always pass a block that gets evaluated in the context of the instance at usage time, so that you can write regular instance-method code. This also removes the need of having to be mindful about the format for values (e.g: block or symbol?), and the need of having to wrap most values with a block.

Finally, such approach would also enable supporting other options in the future. Not that we want these now, but think in [jbuilder's](https://github.com/rails/jbuilder) nested properties or `array!` helpers.